R/utilites.R

```diff
@@ -287,3 +290,23 @@ IsQTable <- function(x)
     inherits(x, "qTable") ||
     (!is.null(attr(x, "questions")) && !is.null(attr(x, "name")))
 }
+
+#' Specify that a string was generated by a Displayr control
+#'
+#' Add an attribute to a character vector called is.control
+#' which can tell other functions that a string originally
+#' came from a combo box or listbox control.
+#' @param control.output A character vector to which the attribute
+#' should be attached.
+#'
+#' @export
+SetIsStringIsFromControl <- function(control.output) {
+    attr(control.output, "is.control") <- TRUE
+    control.output
+}
+
+#' Check whether a character vector was generated by a Displayr control.
+#' noRd
+getIsStringFromControl <- function(x) {
+    return(!is.null(attr(x, "is.control")) && attr(x, "is.control"))
+}
```

Comment:
Can check the attribute a bit more clearly in the code with the `isTRUE` function.
```suggestion
getIsStringFromControl <- function(x) {
    isTRUE(attr(x, "is.control"))
}
```